File: src/main/java/red/tetracube/slurpanize/baker/foodlab/payloads/FoodlabFoundryRequest.java
package red.tetracube.slurpanize.baker.foodlab.payloads;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

public class FoodlabFoundryRequest {

    @NotNull
    @NotEmpty
    @JsonProperty("foodlabName")
    public String name;

    @NotNull
    @NotEmpty
    @JsonProperty("administratorUsername")
    public String username;

    @NotNull
    @NotEmpty
    @JsonProperty("administratorPassword")
    public String password;

}

File: src/main/kotlin/tetracubered/slurpanizeBaker/foodlabFoundry/payloads/FoodlabFoundryRequest.kt
package tetracubered.slurpanizeBaker.foodlabFoundry.payloads

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotNull

data class FoodlabFoundryRequest @JsonCreator constructor(
    @field:NotNull
    @field:NotEmpty
    @JsonProperty("foodlabName")
    val name: String? = null,

    @field:NotNull
    @field:NotEmpty
    @JsonProperty("administratorUsername")
    val username: String? = null,

    @field:NotNull
    @field:NotEmpty
    @JsonProperty("administratorPassword")
    val password: String? = null
)